text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,555 @@
+            2. For each `KubectlVX_YLayer` class, we will include the latest
+               available version of Helm which is compatible with that version
+               of Kubernetes.
+            3. A weekly automated task will update the minor and patch versions
+               of helm within the supported ranges.
+            4. The automated task that checks for new supported minor versions
+               of kubectl, will also attempt to figure out what is the latest
+               version of helm compatible with that version of kubectl. This
+               will be verified by a human in PR review to make sure it is
+               correct.
+    2. `@aws-cdk/lambda-layer-aws-cli` - Major versions 1 and 2 will both be
+       available, and automatically updated to the latest minor or patch
+       version.
+        1. This package will have two APIs: `AwsCliLayer`, and `AwsCliV2Layer`.
+        2. Automatically check for new minor and patch versions from Pypi -
+           [awscli](https://pypi.org/project/awscli/),
+           [awscliv2](https://pypi.org/project/awscliv2/). Each update will
+           result in a minor version bump of this package.
+    3. `@aws-cdk/lambda-layer-node-proxy-agent` - New minor versions will
+       automatically be picked up and available with a minor version bump of the
+       library.
+        1. This can be implemented with `npm-check-updates`, since
+           `node-proxy-agent` is an npm package.
+    4. AWS CDK v2 will automatically pick up new minor versions of the above
+       libraries.
+
+After the above is implemented, this mechanism can be extended to include
+additional Lambda Layers with different dependencies. The solution should be
+implemented in such a way that new dependencies can be added with the below
+steps.
+1. Add a new directory in the `aws-cdk` monorepo: `@aws-cdk/lambda-layer-MyNewDependency`.
+2. Implement the MyNewDependencyLayer API.
+3. If the dependency is from npm, then updates will happen automatically. If it
+   is from some other source, then extend the automation
+   from the above steps to also update the new dependency.
+
+Once this solution is implemented, it will be possible to write more concrete
+steps. And those will be included in the Contributing Guide.
+
+##### What about customers executing their CDK apps in network-restricted environments?
+
+This question raises an important drawback of this solution. It will result in a
+breaking change for customers who are using these Lambda Layers in
+network-restricted environments. If customers are able to use the CDK CLI in
+their network-restricted environments, then it is reasonable to assume they were
+able to acquire the `aws-cdk` package from npm somehow. This implies customers
+will be able to acquire the `@aws-cdk/lambda-layer-X` packages from npm. There
+are a few options for these customers:
+1. If using TypeScript or JavaScript, add each necessary package the
+   dependencies of their CDK app.
```

Comment:
[Above](https://github.com/aws/aws-cdk-rfcs/pull/430/files#diff-94270ba514d2edadf1d22d85d22900af665e6aa9cbc8ecbea733105b83d39d35R219-R221) it says that we would automatically install the lambda layer dependencies. Why are we explaining how customers can go install these manually here?

Reply:
This is for customers using the CDK in environments where we *can't* automatically download these for them. So we need to provide a manual workaround for them. Let me know if that's not clear from the doc!